### PR TITLE
Add block types report

### PR DESCRIPTION
**Pull request by jhonatan-lopes**
# Description

Adds a "Block types" report to the Wagtail admin.

~~:warning: Depends on #11024 !! :warning:~~
~~:warning: Rebase before merging! :warning:~~
:heavy_check_mark: rebasing done!

Link to sample test page: `cms/reports/block-types-report`
Related PRs/issues: #10855 

# Checklist

**Tests**
- [X] Is the code I'm adding covered by tests?

**Changes in Models:**
- ~~[ ] Did I update or add new fake data?~~
- ~~[ ] Did I squash my migration?~~
- ~~[ ] Are my changes [backward-compatible](https://github.com/mozilla/foundation.mozilla.org/blob/main/docs/workflow.md#django-migrations-what-to-do-when-working-on-backward-incompatible-migrations). If not, did I schedule a deploy with the rest of the team?~~

**Documentation:**
- ~~[ ] Is my code documented?~~
- ~~[ ] Did I update the READMEs or wagtail documentation?~~

**Merge Method**
**💡❗Remember to use squash merge when merging non-feature branches into `main`**

### CODE REVIEW

File: network-api/networkapi/reports/views.py [hunk 1]
```diff
@@ -93,3 +94,42 @@ def get_queryset(self):
...
+
+class BlockTypesReportView(ReportView):
+    title = "Block types report"
+    template_name = "pages/reports/block_types_report.html"
+    header_icon = "placeholder"
+
+    def decorate_paginated_queryset(self, object_list):
+        # Build a cache map of PageBlock's block name to content types
+        page_blocks = PageBlock.objects.all().prefetch_related("page__content_type")
+        page_blocks_to_content_types = {pb.block: [] for pb in page_blocks}
```
Comment (victoriachan) [inline network-api/networkapi/reports/views.py:107]: You could use defaultdict ([see example](https://docs.python.org/3/library/collections.html#defaultdict-examples)) to skip the dictionary comprehension.

e.g. 

```
from collections import defaultdict
...

page_blocks_to_content_types = defaultdict(list)
```

**Review by victoriachan — CHANGES_REQUESTED** [review]
Hi @jhonatan-lopes, 

This looks great and works well when I tested. I have a few suggestions, mostly to improve code clarity.

Also while poking around, I noticed that the page count includes unpublished pages, and draft changes. Is this obvious to the admins and editors using this report? 

Also, if a page is created via 'copy', the stats are not updated until the `./manage.py block_inventory` command is run. Just checking that you have plans to add this command as a scheduled task?

File: network-api/networkapi/reports/templatetags/report_tags.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    truncated = []
+    count_remaining = 0
+    if len(content_types) > 3:
+        truncated = content_types[3:]
+        content_types = content_types[:3]
+        count_remaining = len(truncated)
+    return {
+        "content_types_truncated": content_types,
+        "content_types_remaining": truncated,
+        "count_remaining": count_remaining,
```
[the remark this answers is not in the document]
Comment (jhonatan-lopes) [reply network-api/networkapi/reports/templatetags/report_tags.py:41]: Great catch, thanks! This was confusing indeed

File: network-api/networkapi/reports/views.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jhonatan-lopes) [reply network-api/networkapi/reports/views.py:107]: Great point :+1:

File: network-api/networkapi/reports/views.py [hunk 3]
```diff
@@ -93,3 +94,42 @@ def get_queryset(self):
...
+        for page_block in page_blocks:
+            if page_block.page.content_type not in page_blocks_to_content_types[page_block.block]:
+                page_blocks_to_content_types[page_block.block].append(page_block.page.content_type)
+
+        # Get the content_types for each block name
+        for page_block in object_list:
```
[the remark this answers is not in the document]
Comment (jhonatan-lopes) [reply network-api/networkapi/reports/views.py:113]: I've renamed it to `block_report_item` (KISS)

File: network-api/networkapi/reports/views.py (hunk 1 above)
Comment (jhonatan-lopes) [reply network-api/networkapi/reports/views.py:107]: Thanks! TIL :mortar_board:

**Comment by jhonatan-lopes** [comment]
Thanks @victoriachan! Awesome feedback, especially the draft page counting. I had not thought about that but I've changed the query now to only include live pages! This is now ready for a new pass-through

**Review by victoriachan — APPROVED** [review]
Looks good!

**Pull request merged**